niceman/utils.py

```diff
@@ -1014,6 +986,50 @@ def get_cmd_batch_len(arg_list, cmd_len):
     return max((_MAX_LEN_CMDLINE - cmd_len) // (max_len + 1), 1)
 
 
+def execute_command_batch(session, command, args, expected_exceptions=None):
+    """
+    Generator that executes session.execute_command, with batches of args
+
+    We want to call commands like "apt-cache policy" on a large number of
+    packages, but risk creating command-lines that are too long. This
+    function is a generator that will call execute_command but with batches
+    of arguments (to stay within the command-line length limit) and yield the
+    results.
+
+    Parameters
+    ----------
+    session
+      Session object that implements the execute_command() member
+    command : array
```

Comment:
I think `sequence` (preferably) or `list` should be used rather than
`array` (here and a few other places) to avoid confusion with arrays
from the stdlib or numpy.

Reply:
I started with "list" but then pycharm threw some PEP errors with my use of len().  I'll try sequence...

Reply:
Done.